ibis/backends/tests/test_generic.py

```diff
@@ -1028,9 +1028,9 @@ def test_pivot_longer(backend):
         param(
             "first",
             marks=pytest.mark.notimpl(
-                ["trino"],
+                ["trino", "snowflake"],
                 raises=AssertionError,
-                reason="trino is more arbitrary than other backends",
+                reason="backend is more arbitrary than other backends",
```

Comment:
Does this mean that `pivot_longer` works and is correct, but the row output/ordering isn't consistent so the test sometimes fails? If so, perhaps we should change the test to not error, since right now it looks like `pivot_longer` isn't implemented for these backends.

Reply:
Not a blocker of course, if this fixes the flaky test feel free to merge.

Reply:
> Does this mean that `distinct` works and is correct, but the row output/ordering isn't consistent so the test sometimes fails? If so, perhaps we should change the test to not error, since right now it looks like `distinct` isn't implemented for these backends.

Yeah, unfortunately arbitrary really is arbitrary, so this is hard to compare against pandas, because snowflake's any_value keeps nulls, so we use min in our implementation to workaround that.

Reply:
In these cases, could we just check that the guarantees of `distinct` are true (the rows are distinct on the proper columns) and not compare against a single "correct" solution?

Reply:
I'll move it to just a length check for now.